### ISSUE TITLE
Moving fieldset in Bard deletes text

### ISSUE THREAD

**Issue by vladdu** [statamic/v2-hub#1714]
### Steps to reproduce
1. Create a page with a bard field with a fieldset containing text and image (just like the one used in the docs). 
2.  Edit the page with one of each, add some values to the fields.
3.  In the text field, mark everything as bold, then make a part in the middle as normal text again
4. Move the image by dragging it over the text: first, the text filed is split where the normal text is, so you can drop the image in the middle; but when doing it, the normal text is deleted. In the yaml, it is replaced with &nbsp;

I don't have a strong opinion on if the text should be possible to split like that (I tend to think 'no'), but its contents should definitely not be deleted.

If I make the text field a H2 and then back to a regular text, then the problem disappears because the content is now wrapped in <p></p>. So I suppose the issue is in the bard editor, if the text is not inside a html tag, the DOM gets messed up.

I suppose the content of the text should get wrapped in a paragraph from the start, if it is a top-level text fieldset?

### Server Details

**Operating System**: Ubuntu 17.10
**Web Server:** Apache 2.4
**PHP Version:** 7
**Statamic Version:** 2.8
**Updated from an older Statamic or fresh install:** updated

**Comment by jasonvarga** [comment]
Can you record a video of this?

**Comment by vladdu** [comment]
https://www.youtube.com/watch?v=9xEw_GlQm_A

A related issue I noriced: If one presses </>, the saved value of the text is wrapped in <p> and the <b> tags have even `style="font-name: whatever"`.

**Comment by vladdu** [comment]
I see now that the docs say that the text fields are saved as HTML, so I guess my above comment is correct. So I guess that when the CP form is initialized, it should check if there are p tags around the text and add them.

In a related vein, would it be possible to save the text as Markdown, with an option?

**Comment by jasonvarga** [comment]
What was the original HTML?

You can already save as markdown by adding `markdown: true`

**Comment by vladdu** [comment]
Originally there was no HTML, just a text string.

Sorry, I expressed myself wrong: I meant, use markdown and keep the toolbar and rendered formatting. So a `save_as_markdown` option for the regular mode. The full markdown mode allows for more advanced formatting that would be difficult to do inline.

**Comment by jasonvarga** [comment]
I mean add `markdown: true` to the Bard config.

```
story:
  type: bard
  markdown: true
  sets: ...
```

**Comment by vladdu** [comment]
Yes, I did that and the text paragraphs became pure text. The toolbar disappears as the docs say and the markdown is shown raw, not rendered. Which is kind of the purpose of Bard. On the other hand, the markdown option is experimental.

**Comment by jasonvarga** [comment]
We decided that if you're choosing Markdown, you're choosing it for the writing experience. You aren't gaining much by just _saving_ markdown. It's experimental until we can bring back a toolbar and maybe a couple of other things.

**Comment by vladdu** [comment]
Ok, fair enough. But if the initial value of a paragraph is not wrapped in p tags, it should be done automatically, then the original issue gets irrelevant, as the text will never be naked.
